### PR TITLE
Cem 2372 - create delivery radius component

### PR DESCRIPTION
**Pull request by roggc**
deliveryRadius4

### CODE REVIEW

File: src/Containers/DeliveryRadius/DeliveryRadius.tsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+interface IMapCoordinates{
+    lat:number;
+    lng:number;
+}
+
+export interface IDeliveryRadiusProps{
+    title:string;
+    description:string;
+    width:number;
```
Comment (ralph-dev) [inline src/Containers/DeliveryRadius/DeliveryRadius.tsx:37]: Width of what the entire component? Or SliderWidth?

File: src/Containers/DeliveryRadius/DeliveryRadius.tsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    unit:DistanceUnit;
+    mapCoordinates:IMapCoordinates;
+    mapZoom:number;
+}
+
+export const DeliveryRadius:React.FC<IDeliveryRadiusProps>=({width,title,description,leftMarkContent,rightMarkContent,sliderProps,unit,mapCoordinates,mapZoom}):React.ReactElement=>{
+    const [sliderValue,setSliderValue]=useState(sliderProps?.min??0)
+
+    const mapContainer=useRef<HTMLDivElement>(null)
+    const mapApikey='gSSgDhU5omF7RhwKqsy_EenuqNgG24F9pIRck2Dkiu0'
```
Comment (ralph-dev) [inline src/Containers/DeliveryRadius/DeliveryRadius.tsx:50]: Isn't this a STATIC variable

File: src/Containers/DeliveryRadius/DeliveryRadius.tsx [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    leftMarkContent:string;
+    rightMarkContent:string;
+    sliderProps?:ISliderProps;
+    unit:DistanceUnit;
+    mapCoordinates:IMapCoordinates;
+    mapZoom:number;
+}
+
+export const DeliveryRadius:React.FC<IDeliveryRadiusProps>=({width,title,description,leftMarkContent,rightMarkContent,sliderProps,unit,mapCoordinates,mapZoom}):React.ReactElement=>{
+    const [sliderValue,setSliderValue]=useState(sliderProps?.min??0)
```
Comment (ralph-dev) [inline src/Containers/DeliveryRadius/DeliveryRadius.tsx:47]: Spacing around ??

File: src/Containers/DeliveryRadius/DeliveryRadius.tsx [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    )
+}
+
+interface IRootContainerProps{
+    width:number;
+}
+
+const RootContainer=styled.div<IRootContainerProps>`
+border-radius:${Theme.dimensions.radius};
+${({width}):string=>`width:${width}px;`}
```
Comment (ralph-dev) [inline src/Containers/DeliveryRadius/DeliveryRadius.tsx:113]: What??

Comment (ralph-dev) [reply src/Containers/DeliveryRadius/DeliveryRadius.tsx:113]: Use props so no repetition

**Review by ralph-dev — CHANGES_REQUESTED** [review]
Code Quality, the code is looking much better than your before code btw your improving quickly!!

File: src/Containers/DeliveryRadius/DeliveryRadius.tsx (hunk 4 above)
Comment (roggc) [reply src/Containers/DeliveryRadius/DeliveryRadius.tsx:115]: I don't understand this, sorry

Comment (ralph-dev) [reply src/Containers/DeliveryRadius/DeliveryRadius.tsx:115]: width: ${({width}] => width} px

**Comment by ralph-dev** [comment]
> deliveryRadius2
> 
> 

You need to zoom in and out of the map as the radius grows / shrinks btw

File: src/Utils/Hooks/useMap.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+/**
+ * puts a marker in a map in certain coordinates
+ * @param map {MutableRefObject<H.Map | undefined>} - the map instance reference
+ * @param mapCoordinates {ICenterMap} - the coordinates on the map where to place the mark
+ * @param element {React.ReactElement} - the rendered react element which is the mark
+ */
+export const useMapMarker=(map:MutableRefObject<H.Map | undefined>,mapCoordinates:ICenterMap,element:React.ReactElement):void=>{
+    useEffect(()=>{
```
Comment (ralph-dev) [inline src/Utils/Hooks/useMap.ts:59]: Why not use layouts effect?

**Review by ralph-dev — CHANGES_REQUESTED** [review]
Still some work needed

File: src/Utils/Hooks/useMap.ts (hunk 5 above)
Comment (roggc) [reply src/Utils/Hooks/useMap.ts:59]: I tryied and there is no much difference in the end result so I suppose I will stay with useEffect

File: src/Utils/Hooks/useMap.ts [hunk 6]
```diff
@@ -38,6 +39,10 @@ export const useMap=(mapContainer:React.RefObject<HTMLDivElement>,apikey:string,
                 }
             )
 
+            const ui = H.ui.UI.createDefault(mapInstance.current, defaultLayers)
+            ui?.removeControl('mapsettings')
```
Comment (ralph-dev) [inline src/Utils/Hooks/useMap.ts:43]: The Zoom should be controlled by the Radius not manually. Check Here Code for Delta ;)

**Review by ralph-dev — CHANGES_REQUESTED** [review]
Almost there

File: src/Utils/Hooks/useMap.ts (hunk 6 above)
Comment (roggc) [reply src/Utils/Hooks/useMap.ts:43]: what is Delta?

**Comment by ralph-dev** [comment]
Video?

File: src/Containers/DeliveryRadius/DeliveryRadius.tsx [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+                            marks={{
+                                [sliderProps?.min||0]:<Paragraph size={SLIDER_MARK_SIZE} bold>{leftMarkContent}</Paragraph>,
+                                [sliderProps?.max||100]:<Paragraph size={SLIDER_MARK_SIZE} bold>{rightMarkContent}</Paragraph>,
+                            }}
+                            dotStyle={{
+                                width:0,
+                                height:0,
+                                border:0
+                            }}
+                            step={sliderProps?(sliderProps.max-sliderProps.min)/100:1}
```
Comment (ralph-dev) [inline src/Containers/DeliveryRadius/DeliveryRadius.tsx:108]: 100 magic number

File: src/Utils/Hooks/useMap.ts [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+ * @param apikey {string} - the api key for the sdk
+ * @param center {ICenterMap} - the map coordinates (its center)
+ * @param zoom {number} - the level of zoom applied
+ * @returns {MutableRefObject<H.Map | undefined>} a reference to the map instance created
+ */
+export const useMap=(mapContainer:React.RefObject<HTMLDivElement>,apikey:string,center:ICenterMap,zoom:number):MutableRefObject<H.Map | undefined>=> {
+    const mapInstance=useRef<H.Map>()
+    const platform=useRef<H.service.Platform>()
+    
+    useLayoutEffect(()=>{
```
Comment (ralph-dev) [inline src/Utils/Hooks/useMap.ts:22]: Comment

File: src/Utils/Hooks/useMap.ts [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+ * puts a marker in a map in certain coordinates
+ * @param map {MutableRefObject<H.Map | undefined>} - the map instance reference
+ * @param mapCoordinates {ICenterMap} - the coordinates on the map where to place the mark
+ * @param element {React.ReactElement} - the rendered react element which is the mark
+ * @returns {React.MutableRefObject<H.map.DomMarker | undefined>} a reference to the marker
+ */
+export const useMapMarker=(map:MutableRefObject<H.Map | undefined>,mapCoordinates:ICenterMap,element:React.ReactElement):React.MutableRefObject<H.map.DomMarker | undefined>=>{
+    const marker=useRef<H.map.DomMarker>()
+
+    useLayoutEffect(()=>{
```
Comment (ralph-dev) [inline src/Utils/Hooks/useMap.ts:64]: Comments

**Review by ralph-dev — CHANGES_REQUESTED** [review]
Missing comments

File: src/Utils/Hooks/useMap.ts [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+ * renders a circle in a map centered in some specified coordinates and with some specified radius
+ * @param map {MutableRefObject<H.Map | undefined>} - the map instance reference
+ * @param mapCoordinates {ICenterMap} - the coordinates in the map of the center of the circle
+ * @param value {number} - the radius of the circle
+ * @param unit {number} - the number of meters of the unit used in value
+ * @returns {React.MutableRefObject<H.map.Circle | undefined>} a reference to the circle
+ */
+export const useMapCircle=(map:MutableRefObject<H.Map | undefined>,mapCoordinates:ICenterMap,value:number,unit:number):React.MutableRefObject<H.map.Circle | undefined>=>{
+    const circle=useRef<H.map.Circle>()
+    useLayoutEffect(()=>{
```
Comment (ralph-dev) [inline src/Utils/Hooks/useMap.ts:92]: Comment.

**Review by ralph-dev — CHANGES_REQUESTED** [review]

**Review by ralph-dev — APPROVED** [review]
LGTM but before merging ask yourself
1) Can you edit the SliderKnob Style through Props
2) Can you edit the Map through Props?
If the answer is no, expose what you need then merge.

**Comment by ralph-dev** [comment]
Good to merge

**Pull request merged**